### ISSUE TITLE
Document correct aria usage and examples for our badge component

### ISSUE THREAD

**Issue by Febakke** [digdir/designsystem#2749]
Often when using badges you are giving visual cues that something has happened. There is a new message or an update to an element in the user interface. How should screen readers handle this? We need to document some cases in our storybook.

Suggestion @eirikbacker 
>   - Should have `role="img"` and make `aria-label` prop required, and print content as `::before { content: attr(aria-label) }` instead of using children

## relevant links
[W3 pattern](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA22#example-2-updating-the-shopping-cart-status)

**Comment by mrosvik** [comment]
Will be solved by #3560
